### PR TITLE
Create the updated 'No results' and 'Server timeout' sections

### LINKED ISSUE #602: No results page
## Problem
We need to show a friendly message when there are no results. The page should encourage people to try a different search or browse on the additional source sites.

## Description
The mockup is available in the [Releases file](https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A6869) as both desktop and mobile. I will add the image as an exportable asset ASAP.

|Desktop|Mobile
|:---|:---|
|imagen||

## Additional context
On desktop, the source site is a border version of the [secondary button](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=558%3A2286), while on mobile is the same as desktop but full-width and `16px` top and bottom padding. I was not sure about adding these elements as components since are used only on this page. Any ideas on this?

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #602 by @panchovm 

## Description

This PR creates the redesigned 'No results' and 'Server timeout' pages and the components necessary for the same.

## Testing Instructions

0. Check out the PR and run the dev server.
1. Search for random string like [Poopy-di scoop](https://genius.com/14447294). The server error page is a bit harder to test but it uses the same components so it should work in theory.
2. See the new design.
3. You can also see these designs in the Storybook.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I'm going to put this aside till #530 is merged to avoid merge conflicts.

**Comment by zackkrida** [comment]
@dhruvkb how do you feel about resuming this work? I don't think it's the highest-priority feature, but there are very few merge conflicts here. If the remaining work doesn't feel too difficult it might be good to resume.

**Comment by dhruvkb** [comment]
@zackkrida sorry this just fell out of my head! I'll quickly bring it to a review-able state.

**Comment by zackkrida** [comment]
@dhruvkb no worries! i think it made sense to pause this for a while, so i assumed it was intentional 😅

**Comment by zackkrida** [comment]
Tried to test the server error with the following:

```
API_URL=fake pnpm dev
```

This resulted in a blank page on SSR loads:

CleanShot 2022-03-03 at 07 09 28@2x

And the "No results" view on client loads:

CleanShot 2022-03-03 at 07 09 40@2x

I'm not necessarily suggesting that we fix this specific case in this PR, but maybe that we need a new issue to review all potential API errors and how they're handled generally.

File: src/utils/attribution-html.js [hunk 1]
```diff
@@ -32,4 +32,29 @@ function getAttributionHtml(media, licenseUrl, fullLicenseName) {
...
+  let creator = mediaItem.creator
+  if (mediaItem.creator_url) {
+    creator = `<a href="${mediaItem.creator_url}">${creator}</a>`
+  }
+
+  let license = `${mediaItem.license} ${mediaItem.license_version}`
+  license = `<a href="${mediaItem.license_url}">${license.toUpperCase()}</a>`
+
+  return `${title} by ${creator} is licensed under ${license}.`
+}
```
Comment (zackkrida) [inline src/utils/attribution-html.js:58]: We probably want to use `target="_blank"` and `rel="noopener noreferrer"` on these links. The added complexity might warrant a small helper, something similar to this:

```ts
const buildAnchorTag = (text, href) => 
  `<a target="_blank" rel="noopener noreferrer" href=${href}>${text}</a>`
```

File: src/locales/en.json [hunk 2]
```diff
@@ -619,6 +619,17 @@
   "modal": {
     "close": "Close"
   },
+  "error-images": {
+    "depressed-musician": "A depressed pianist rests their head in their hands.",
+    "waiting-for-a-bite": "Three boys sit on a broken log while two of them fish."
+  },
+  "no-results": {
+    "heading": "We couldn't find anything for \"{query}\".",
+    "alternatives": "Try a different search or see the additional sources."
```
Comment (zackkrida) [inline src/locales/en.json:628]:
```suggestion
    "alternatives": "Try a different search or see the additional sources."
```

Something about "see the additional sources" doesn't seem quite right to me, but I'm struggling to come up with alternative text.

File: src/constants/errors.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const NO_RESULT = 'NO_RESULT'
```
Comment (zackkrida) [inline src/constants/errors.js:1]: Could we get a top-level comment about the intentions of this file? Is it intended for _any_ error in the application or errors specific to media fetching?

File: src/assets/error_images/image_info.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        "license_version": "1.0",
+        "license_url": "https://creativecommons.org/publicdomain/zero/1.0/"
+      }
+    },
+    {
+      "error": "SERVER_TIMEOUT",
+      "image": {
+        "id": "waiting-for-a-bite",
+        "file": "waiting_for_a_bite",
+        "title": "Waiting for a Bite, from Harper's Weekly, August 22, 1874, p. 693",
```
Comment (krysal) [inline src/assets/error_images/image_info.json:21]: When is this image used?

File: src/components/VErrorImage/VErrorSection.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<template>
```
Comment (krysal) [inline src/components/VErrorImage/VErrorSection.vue:1]: Perhaps the folder for these files should be renamed to "VErrorSection"? As it's not only about the error image.

File: src/components/VErrorImage/VErrorImage.vue [hunk 6]
```diff
@@ -0,0 +1,58 @@
+<template>
+  <figure class="error-image">
+    <img :src="image.src" :alt="$t(image.alt)" :title="$t(image.alt)" />
```
[the remark this answers is not in the document]
Comment (zackkrida) [reply src/components/VErrorImage/VErrorImage.vue:3]: Nice find!

File: src/locales/en.json (hunk 2 above)
Comment (sarayourfriend) [reply src/locales/en.json:628]: "Try a different query or use one of the additional sources to expand your search." Maybe something along those lines?

File: src/components/VErrorImage/meta/VErrorSection.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+## No results
+
+This section appears when a search returns no hits.
+
+export const NoResultsTemplate = (args, { argTypes }) => ({
+  template: `
+    <VErrorSection>
+      <template #image>
+       <VErrorImage :error-code=errorCode />
```
Comment (sarayourfriend) [inline src/components/VErrorImage/meta/VErrorSection.stories.mdx:32]:
```suggestion
       <VErrorImage :error-code="errorCode" />
```

File: src/components/VErrorImage/meta/VErrorSection.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+export const ServerTimeoutTemplate = (args, { argTypes }) => ({
+  template: `
+    <VErrorSection>
+      <template #image>
+        <!-- On large screens, image is in the left pane. -->
+        <VErrorImage class="hidden md:block" :error-code=errorCode />
+      </template>
+      <VServerTimeout/>
+      <!-- On small screens, image is below error message. -->
+      <VErrorImage class="md:hidden mt-10" :error-code=errorCode />
```
Comment (sarayourfriend) [inline src/components/VErrorImage/meta/VErrorSection.stories.mdx:76]:
```suggestion
        <VErrorImage class="hidden md:block" :error-code="errorCode" />
      </template>
      <VServerTimeout/>
      <!-- On small screens, image is below error message. -->
      <VErrorImage class="md:hidden mt-10" :error-code="errorCode" />
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! I don't think the suggestions I left in the storybook are necessary as it appears to work fine when I run it locally (but definitely a TIL for me that Vue will allow you to pass variables in the templates without quoting them...)

File: src/utils/attribution-html.js (hunk 1 above)
Comment (dhruvkb) [reply src/utils/attribution-html.js:58]: What are CC's policies on attribution language? Can you attribute in any language? Are there any rules that the atribution must be understood by the viewers? If I run a blog in English and my audience is largely English-speaking, do I necessarily have to attribute media I use in English?

File: src/constants/errors.js (hunk 3 above)
Comment (dhruvkb) [reply src/constants/errors.js:1]: This is for any error "fatal" that might arise i.e. any error that leads to the content being replaced by the image + explanation view.

File: src/components/VErrorImage/VErrorImage.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (dhruvkb) [reply src/components/VErrorImage/VErrorImage.vue:3]: @krysal the alt text is the same as the title. The error images do not have a `title` property.

File: src/assets/error_images/image_info.json (hunk 4 above)
Comment (dhruvkb) [reply src/assets/error_images/image_info.json:21]: This is the image to show when an API request times out for a search query.

File: src/locales/en.json (hunk 2 above)
Comment (dhruvkb) [reply src/locales/en.json]: I like the first suggestion too, it's clear but not too verbose. Fixed in 1d33193b.

File: src/components/VErrorImage/meta/VErrorSection.stories.mdx (hunk 7 above)
Comment (dhruvkb) [reply src/components/VErrorImage/meta/VErrorSection.stories.mdx]: Even if it works, the quotes are not supposed to be omitted so this is a good catch, thanks! Fixed in 9129094.

File: src/components/VErrorImage/meta/VErrorSection.stories.mdx (hunk 8 above)
Comment (dhruvkb) [reply src/components/VErrorImage/meta/VErrorSection.stories.mdx]: Fixed in 9129094.

File: src/components/VErrorImage/VErrorSection.vue (hunk 5 above)
Comment (dhruvkb) [reply src/components/VErrorImage/VErrorSection.vue]: This was on my mind as a todo, right till I pushed the PR. Thank you so much for catching it. Renamed in 2599478.

File: src/constants/errors.js (hunk 3 above)
Comment (dhruvkb) [reply src/constants/errors.js:6]: Added an explanatory comment to the file in 435c3bb.

**Review by zackkrida — APPROVED** [review]
lgtm.

**Pull request merged**